integrations/salesforce/subscription.go

```diff
@@ -0,0 +1,143 @@
+package salesforce
+
+import (
+	"context"
+	"sync"
+
+	pb "github.com/developerforce/pub-sub-api/go/proto"
+	"github.com/linkedin/goavro/v2"
+	"go.uber.org/zap"
+
+	// "go.autokitteh.dev/autokitteh/integrations/common"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+var (
+	// Key = Salesforce instance URL (to ensure one gRPC client per app).
+	// Writes happen during server startup, and when a user
+	// creates a new Salesforce connection in the UI.
+	pubSubClients = make(map[string]*pb.PubSubClient)
+
+	mu = &sync.Mutex{}
+)
+
+func (h handler) Subscribe(instanceURL, orgID, accessToken string, cid sdktypes.ConnectionID) {
+	// Ensure multiple users don't reference the same app at the same time.
+	mu.Lock()
+	defer mu.Unlock()
+
+	if _, ok := pubSubClients[instanceURL]; ok {
+		return
+	}
+
+	ctx := context.Background()
+	conn, err := initConn(accessToken, instanceURL, orgID, h.logger)
+	if err != nil {
+		h.logger.Error("failed to create gRPC connection", zap.Error(err))
+		return
+	}
+
+	client := pb.NewPubSubClient(conn)
+	pubSubClients[instanceURL] = &client
+
+	for _, eventType := range supportedEventTypes() {
+		go h.handleSalesForceSubscription(ctx, client, eventType, cid)
+	}
+}
+
+func (h handler) handleSalesForceSubscription(ctx context.Context, client pb.PubSubClient, topicName string, cid sdktypes.ConnectionID) {
+	const defaultBatchSize = int32(100)
+	// var latestReplayId []byte
+	numLeftToReceive := 0
+
+	stream, err := client.Subscribe(ctx)
+	if err != nil {
+		h.logger.Error("failed to create stream", zap.Error(err))
```

Comment:
For easier triage/debugging in Datadog:
```suggestion
		h.logger.Error("failed to create gRPC stream for Salesforce events", zap.Error(err))
```